cpp/src/parquet/types.cc

```diff
@@ -244,14 +244,13 @@ SortOrder::type DefaultSortOrder(Type::type primitive) {
     case Type::BOOLEAN:
     case Type::INT32:
     case Type::INT64:
+    case Type::INT96:
     case Type::FLOAT:
     case Type::DOUBLE:
       return SortOrder::SIGNED;
     case Type::BYTE_ARRAY:
     case Type::FIXED_LEN_BYTE_ARRAY:
       return SortOrder::UNSIGNED;
-    case Type::INT96:
-      return SortOrder::UNKNOWN;
```

Comment:
This change appeared untested in the prior iteration of the code; @majetideepak I think this is the correct change but please confirm

Reply:
The current spec (https://github.com/apache/parquet-format/blob/master/src/main/thrift/parquet.thrift#L821) says it is undefined. So this change is incorrect.

Reply:
What should be the default comparator, then? Prior to this patch the default was SIGNED

Reply:
Actually, I don't think this is true. I'm struggling to figure out how Int96 statistics were working at all prior to this patch. Where is the Int96 comparator created? Other types go through `Comparator::Make` but that does not support Int96

https://github.com/apache/arrow/blob/2a4e48e384be618fc476e8ba8091642c7091f705/cpp/src/parquet/util/comparison.cc#L28

Reply:
We don't create the `TypedStats` object at all for INT96. We don't compute statistics as a result and we never invoke `Comparator::Make` for INT96.`

https://github.com/apache/arrow/blob/2a4e48e384be618fc476e8ba8091642c7091f705/cpp/src/parquet/column_writer.cc#L647-L649

Reply:
Ah OK. I will fix. Thanks

Reply:
Fixed